### PR TITLE
Add member page 

### PR DESCRIPTION
**Pull request by rubasider**
in the client side 4 text fields were add to take the employee's data and then add this data to the database and i used sweet alert  to pop up messages in case of success or error
and there is a  file called SwitchRoute is used to switch between the endpoints 
to view the page use   /dashboard

### CODE REVIEW

File: server/controller/employee/getEmployees.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// const getEmployeesBy = require('../../database/query/employee/getEmployees');
```
Comment (mantagen) [inline server/controller/employee/getEmployees.js]: Why is this all commented out?

File: server/controller/employee/addNewEmployee.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+const postNewEmployee = require('../../database/query/employee/postNewEmployee');
+
+const addNewEmployee = (req, res) => {
+	const Empdata = req.body;
```
Comment (mantagen) [inline server/controller/employee/addNewEmployee.js]: `employeeData`

- PascalCase is reserverd for classes in javascript (https://www.robinwieruch.de/javascript-naming-conventions)

File: client/src/Componants/dashboard/NewMember.jsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+		const target = event.target;
+		const form = { ...empData };
+		form[target.name] = target.value;
+		setEmpdata(form);
+	};
+
+	const handleSubmit = (event) => {
+		event.preventDefault();
+		const TheData = { ...empData };
+		console.log('theData', TheData);
```
Comment (mantagen) [inline client/src/Componants/dashboard/NewMember.jsx]: :fire:

File: client/src/Componants/dashboard/NewMember.jsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+				swal('Good job!', 'employee added successfuly', 'success');
+			})
+			.catch((err) => {
+				if (err.response.data)
+					swal('error', err.response.data.message, 'error');
+			});
+	};
+
+	console.log('data', empData);
```
Comment (mantagen) [inline client/src/Componants/dashboard/NewMember.jsx]: :fire:

File: package.json [hunk 5]
```diff
@@ -21,12 +21,15 @@
 	},
 	"homepage": "https://github.com/GSG-K3/360HMR#readme",
 	"dependencies": {
+		"@material-ui/core": "^4.11.0",
```
Comment (mantagen) [inline package.json]: I think this should be in `client/package.json`?

**Review by mantagen — CHANGES_REQUESTED** [review]

File: client/src/App.js [hunk 6]
```diff
@@ -1,23 +1,13 @@
 import React from 'react';
+import SwitchRoute from './Componants/base/SwitchRoute';
```
Comment (yaqootturman) [inline client/src/App.js]: What is this path? did we talk about it? we talked about file structure and I don't think you mentioned this!

Comment (yaqootturman) [reply client/src/App.js]: and please note the components folder is smaller

File: client/src/Componants/dashboard/NewMember.jsx [hunk 7]
```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { TextField, Button } from '@material-ui/core';
+import { makeStyles } from '@material-ui/core/styles';
+import swal from 'sweetalert';
+import axios from 'axios';
+const useStyles = makeStyles((theme) => ({
```
Comment (yaqootturman) [inline client/src/Componants/dashboard/NewMember.jsx]: I don`t prefer using styling include jsx file

**Review by yaqootturman — CHANGES_REQUESTED** [review]

File: client/src/Componants/dashboard/NewMember.jsx (hunk 7 above)
Comment (yaqootturman) [reply client/src/Componants/dashboard/NewMember.jsx]: add the template file, please!

File: package.json [hunk 8]
```diff
@@ -23,10 +23,12 @@
...
 		"concurrently": "^5.2.0",
 		"dotenv": "^8.2.0",
+		"env2": "^2.2.2",
 		"express": "^4.17.1",
 		"nodemon": "^2.0.4",
-		"pg": "^8.2.1",
-		"request": "^2.88.2"
+		"pg": "^7.2.1",
+		"request": "^2.88.2",
+		"sweetalert": "^2.1.2"
```
Comment (yaqootturman) [inline package.json:31]: The sweetalert should not install here! , you should install it in client-side

File: client/src/App.js [hunk 9]
```diff
@@ -1,23 +1,25 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-        </a>
-      </header>
-    </div>
-  );
+	return (
+		<Router>
+			<Grid container>
```
Comment (yaqootturman) [inline client/src/App.js:11]: Is there a reason to make router as a grid?

File: pull_request_template.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
+### What does this PR do?
+
+- Call the add new employee with http://localhost:3000/dashboard endpoint.
+
+* On the dashBoard page you can add new employee tab and add the employee to the
+  database after you add his name,email,date of hireing and position.
+
+* Adding button to confirm adding the employee.
```
Comment (yaqootturman) [inline pull_request_template.md:8]: Who should do this point?

File: server/controller/employee/addNewEmployee.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+
+const addNewEmployee = (req, res) => {
+	const employeeData = req.body;
+
+	postNewEmployee(employeeData)
+		.then(() => {
+			res.status(200).json({ message: 'employee added succesfuly' });
+		})
+		.catch((err) => {
+			console.log('Error in post Employee : ', err);
```
Comment (yaqootturman) [inline server/controller/employee/addNewEmployee.js:11]: delete console

File: server/controller/employee/addNewEmployee.js [hunk 12]
```diff
@@ -0,0 +1,15 @@
+const postNewEmployee = require('../../database/query/employee/postNewEmployee');
+
+const addNewEmployee = (req, res) => {
+	const employeeData = req.body;
```
Comment (yaqootturman) [inline server/controller/employee/addNewEmployee.js:4]: I think you should make sure and validate the data before pass them to the query

**Review by yaqootturman — CHANGES_REQUESTED** [review]

**Pull request closed without merging**